pkg/controllers/qat/controller.go

```diff
@@ -164,6 +183,76 @@ func (c *controller) UpdateStatus(rawObj client.Object, ds *apps.DaemonSet, node
 	return updated, nil
 }
 
+func removeVolume(volumes []v1.Volume, name string) []v1.Volume {
+	newVolumes := []v1.Volume{}
+
+	for _, volume := range volumes {
+		if volume.Name != name {
+			newVolumes = append(newVolumes, volume)
+		}
+	}
+
+	return newVolumes
+}
+
+func setInitContainer(dsSpec *v1.PodSpec, dpSpec devicepluginv1.QatDevicePluginSpec) {
+	yes := true
+
+	qatDeviceDriver := map[string]string{
+		"dh895xccvf": "0434 0435",
+		"c3xxxvf":    "19e2",
+		"c6xxvf":     "37c8",
+		"d15xxvf":    "6f54",
+		"4xxxvf":     "4940",
+		"c4xxxvf":    "18a0",
+	}
+
+	enablingPfPciIDs := make([]string, 0, len(qatDeviceDriver))
+	for _, v := range dpSpec.KernelVfDrivers {
+		enablingPfPciIDs = append(enablingPfPciIDs, qatDeviceDriver[string(v)])
+	}
+
+	dsSpec.InitContainers = []v1.Container{
+		{
+			Image:           dpSpec.InitImage,
+			ImagePullPolicy: "IfNotPresent",
+			Name:            "init-sriov-numvfs",
+			Env: []v1.EnvVar{{
+				Name:  "ENABLING_PF_PCI_IDs",
```

Comment:
I think this needs a better variable name (same for the variables above). How about:

```suggestion
				Name:  "ENABLED_QAT_PF_PCIIDS",
```

Reply:
Done!